packages/teams-js/src/public/video.ts

```diff
@@ -10,46 +10,101 @@ import { runtime } from './runtime';
+
+  /**
+   * Align with the W3C spec: https://www.w3.org/TR/webcodecs/
+   */
+
+  export type AlphaOption = 'discard' | 'keep';
+
+  /**
+   * Align with the W3C spec: https://www.w3.org/TR/webcodecs/
+   */
+  export interface PlaneLayout {
+    offset: number;
+    stride: number;
+  }
+
+  /**
+   * Align with the W3C spec: https://www.w3.org/TR/webcodecs/
+   */
+  export interface VideoFrameCopyToOptions {
+    layout?: PlaneLayout[] | undefined;
+    rect?: DOMRectInit | undefined;
+  }
+
+  /**
+   * Align with the W3C spec: https://www.w3.org/TR/webcodecs/
+   */
+  export interface VideoFrameInit {
+    alpha?: AlphaOption | undefined;
+    displayHeight?: number | undefined;
+    displayWidth?: number | undefined;
+    duration?: number | undefined;
+    timestamp?: number | undefined;
+    visibleRect?: DOMRectInit | undefined;
+  }
+
+  /**
+   * Align with the W3C spec: https://www.w3.org/TR/webcodecs/
+   */
+  export interface VideoFrameBufferInit {
+    codedHeight: number;
+    codedWidth: number;
+    colorSpace?: VideoColorSpaceInit | undefined;
+    displayHeight?: number | undefined;
+    displayWidth?: number | undefined;
+    duration?: number | undefined;
+    format: VideoPixelFormat;
+    layout?: PlaneLayout[] | undefined;
+    timestamp: number;
+    visibleRect?: DOMRectInit | undefined;
+  }
+
+  /**
+   * VideoFrame definition, align with the W3C spec: https://www.w3.org/TR/webcodecs/
    */
   export interface VideoFrame {
-    /**
-     * Video frame width
-     */
-    width: number;
-    /**
-     * Video frame height
-     */
-    height: number;
-    /**
-     * Video frame buffer
-     */
-    data: Uint8ClampedArray;
-    /**
-     * NV12 luma stride, valid only when video frame format is NV12
-     */
-    lumaStride?: number;
-    /**
-     * NV12 chroma stride, valid only when video frame format is NV12
-     */
-    chromaStride?: number;
-    /**
-     * RGB stride, valid only when video frame format is RGB
-     */
-    stride?: number;
-    /**
-     * The time stamp of the current video frame
-     */
-    timestamp?: number;
+    readonly codedHeight: number;
+    readonly codedRect: DOMRectReadOnly | null;
+    readonly codedWidth: number;
+    readonly colorSpace: VideoColorSpace;
+    readonly displayHeight: number;
+    readonly displayWidth: number;
+    readonly duration: number | null;
+    readonly format: VideoPixelFormat | null;
+    readonly timestamp: number | null;
+    readonly visibleRect: DOMRectReadOnly | null;
+    allocationSize(options?: VideoFrameCopyToOptions): number;
+    clone(): VideoFrame;
+    close(): void;
+    copyTo(destination: AllowSharedBufferSource, options?: VideoFrameCopyToOptions): Promise<PlaneLayout[]>;
   }
 
+  /**
+   * VideoFrame definition, align with the W3C spec: https://www.w3.org/TR/webcodecs/
+   */
+  // eslint-disable-next-line no-var, strict-null-checks/all
+  declare var VideoFrame: {
```

Comment:
Out of curiosity, why are we using `declare var` here? Additionally, does this cause any conflicts with the above interface being VideoFrame as well?

Reply:
Line 96 declares a variable named "VideoFrame" and its type, which includes 2 constructors that returns a `VideoFrame` object. By doing so we are able to call "new VideoFrame(...)" in line 359.

Reply:
Line 96 declares a variable named "VideoFrame" and its type, which includes 2 constructors that returns a `VideoFrame` object. By doing so we are able to call "new VideoFrame(...)" in line 359.